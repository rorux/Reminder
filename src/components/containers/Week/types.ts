import { TRecord } from '@store/records/types';

export type TWeekProps = {
  weekArray: Array<null | number>;
  recordList: Array<TRecord>;
};
