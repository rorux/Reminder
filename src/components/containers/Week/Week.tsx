import React from 'react';
import Stack from '@mui/material/Stack';
import { v1 as uuid } from 'uuid';
import { TWeekProps } from './types';
import Day from '@components/Day';
import { TRecord } from '@store/records/types';

const Week: React.FC<TWeekProps> = ({ weekArray, recordList }) => {
  return (
    <Stack direction="row" sx={{ justifyContent: 'space-between', marginBottom: '2px' }}>
      {weekArray.map((number) => {
        const recordsThisDay: Array<TRecord> = [];

        if (number !== null) {
          recordList.forEach((recordObj) => {
            if (recordObj.days.includes(number)) recordsThisDay.push(recordObj);
          });
        }

        return <Day number={number} key={uuid()} records={recordsThisDay} />;
      })}
    </Stack>
  );
};

export default Week;
