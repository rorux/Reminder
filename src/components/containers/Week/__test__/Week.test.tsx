import React from 'react';
import * as redux from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import Week from '../Week';

Enzyme.configure({ adapter: new Adapter() });

const recordList = [
  {
    id: '1',
    title: 'title of monthly remind',
    period: 'monthly',
    monthday: '8',
    holidays: 'afterHoliday',
    days: [1642366800000, 1644872400000],
  },
];

const weekArray = [null, null, 1642366800000];

describe('Week component', () => {
  it('Should render Week component', () => {
    const component = shallow(<Week recordList={recordList} weekArray={weekArray} />);
    expect(component).toMatchSnapshot();
  });
});
