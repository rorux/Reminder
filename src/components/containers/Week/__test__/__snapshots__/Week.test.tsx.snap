// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week component Should render Week component 1`] = `
<ForwardRef(Stack)
  direction="row"
  sx={
    Object {
      "justifyContent": "space-between",
      "marginBottom": "2px",
    }
  }
>
  <Day
    key="b7e92bf0-7165-11ec-a1a3-4d70b950126b"
    number={null}
    records={Array []}
  />
  <Day
    key="b7e95300-7165-11ec-a1a3-4d70b950126b"
    number={null}
    records={Array []}
  />
  <Day
    key="b7e95301-7165-11ec-a1a3-4d70b950126b"
    number={1642366800000}
    records={
      Array [
        Object {
          "days": Array [
            1642366800000,
            1644872400000,
          ],
          "holidays": "afterHoliday",
          "id": "1",
          "monthday": "8",
          "period": "monthly",
          "title": "title of monthly remind",
        },
      ]
    }
  />
</ForwardRef(Stack)>
`;
